feat: qol improvements

harlan-zw opened the pull request.

hannoeru commented on .vitepress/theme/components/Page.vue:
```diff
@@ -19,7 +19,7 @@ import PageFooter from './PageFooter.vue'
 </script>
 
-<style scoped>
+<style scoped lang="scss">
```
You can remove this, also delete scss package.

harlan-zw replied:
Is there a reason not to use `scss`? It's much easier to write

harlan-zw replied:
Removed

hannoeru commented on .vitepress/theme/components/playground/InlinePlayground.vue:
```diff
@@ -316,3 +316,13 @@ onMounted(async() => {
 }
 </style>
+
+<style lang="scss" scoped>
+.inline-playground {
+  @apply mt-8 mb-8;
+  @media (min-width: 1600px) {
+    width: 150%;
+    margin-left: -25%;
+  }
+}
+</style>
```
Personally I don't want to mix different css preprocessor together.

harlan-zw replied:
Normally I would agree, however, because windi is primarily used by sass users (based on issues I have from the nuxt repo), it seems like a good idea to have a repo doing production tests that it's working

I can revert it if you're really against it

harlan-zw replied:
Actually have removed

hannoeru commented on .vitepress/theme/components/sidebar/SideBarLink.ts:
```diff
@@ -14,21 +15,32 @@ export const SideBarLink: FunctionalComponent<{
   const site = useSiteData()
 
-  const headers = route.data.headers
+  const headers = route.data?.headers
```
`route.data` is always have value, if not it will throw error.

hannoeru replied:
It should go to 404 page that you just added.

harlan-zw replied:
@hannoeru There is some weird behaviour when you visit a missing page, `data` is null so any reference to it breaks the page.

See the console and visit a 404 page

hannoeru replied:
Looks like vitepress couldn't handle dynamic route, so my suggestion is delete 404 page & redirect, let netlify handle it.

harlan-zw replied:
I'll just revert all the stuff I did around the redirects & 404 for now

hannoeru commented on netlify.toml:
```diff
@@ -15,4 +15,4 @@
 [[redirects]]
   from = "/*"
   to = "/index.html"
```
should be `/404.html`

harlan-zw commented:
> All colors should also adopt to dark mode (scrollbar, home page v3 article link, etc...)

Yeah sorry about that, should of given my PR a proper review. I've updated it now and it should be looking better.

I have reverted some of my changes to use the css vars you setup. Also reverted the change I did around the transparent scroll bar, I think that looked good

harlan-zw commented on netlify.toml:
```diff
@@ -14,5 +14,5 @@
-  to = "/404.html"
-  status = 404
+  to = "/index.html"
+  status = 301
```
I reverted it to what it originally was, think we can address the 404 issue in a seperate PR.

hannoeru commented on .vitepress/theme/components/playground/InlinePlayground.vue:
```diff
@@ -316,3 +316,13 @@ onMounted(async() => {
 }
 </style>
+
+<style scoped>
+@media (min-width: 1600px) {
+  .inline-playground {
+    @apply mt-8 mb-8;
+    width: 150%;
+    margin-left: -25%;
+  }
+}
+</style>
```
This will break mobile view, and is not only used in homepage.

harlan-zw replied:
this was only for desktop `@media (min-width: 1600px)`, but I'll revert this change as it can have unexpected consequences if not within the content pages

hannoeru commented on .vitepress/config/sidebar.ts:
```diff
@@ -131,25 +131,30 @@ export const sidebar: DefaultTheme.Config['sidebar'] = {
         // { text: 'Nesting CSS', link: '/features/nesting' },
       ],
     },
+    {
+      text: 'Tools',
+      children: [
+        { text: 'Visual Analyzer', link: '/tools/analyzer' },
+      ],
+    },
```
Suggestion: move to end of sidebar.

harlan-zw replied:
Agreed

harlan-zw commented on .vitepress/theme/styles/vars.css:
```diff
@@ -69,11 +70,14 @@
 
 html.dark {
---c-bg: #171717;
+--c-bg: theme('colors.brand.900');
```
Shouldn't they be using the variable? The colour difference between these two is very slight, one is just slightly bluer

harlan-zw replied:
by primary, you mean the `--c-brand`? In that case it's the same as well, just slightly darker to fix the low contrast

harlan-zw replied:
Definitely want to avoid extra work though, do you have any examples where it's messed up?

harlan-zw commented:
I'll just close this off for now @hannoeru  feel free to pull any of the changes if you think there's something useful
